File: back-end/mercado-inteligente-spring/src/main/java/br/edu/usf/mercadointeligentespring/gateways/dto/UserDTO.java
package br.edu.usf.mercadointeligentespring.gateways.dto;

import java.time.LocalDate;
import java.util.List;
import javax.persistence.ElementCollection;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity(name = "User")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class UserDTO {
    @Id
    @GeneratedValue
    private Long id;
    private LocalDate birthDay;
    private String email;
    private String gender;
    private String login;
    private String name;
    private String password;
    @ElementCollection
    private List<String> roles;
}

File: back-end/mercado-inteligente/src/main/kotlin/br/edu/usf/gateways/dto/UserDTO.kt
package br.edu.usf.gateways.dto

import javax.persistence.ElementCollection
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.Id

@Entity(name = "user")
class UserDTO {
    @Id
    @GeneratedValue
    var id: Long? = null;
    lateinit var login: String
    lateinit var name: String
    lateinit var email: String
    lateinit var gender: String
    @ElementCollection
    lateinit var roles: List<String>
    lateinit var password: String
    lateinit var birthDay: String
}